subgraphs/saddle-finance/src/entities/pool.ts

```diff
@@ -56,42 +57,91 @@ import { SimpleRewarder } from "../../generated/templates/Swap/SimpleRewarder";
 export function getOrCreatePool(address: Address): LiquidityPool {
   let pool = LiquidityPool.load(address.toHexString());
   if (!pool) {
-    // Pool was not created through a SwapDeployer
+    // Pool was not created through a SwapDeployer nor Registry
     pool = createPoolFromAddress(address);
   }
   return pool;
 }
 
-export function createPoolFromEvent(event: NewSwapPool): boolean {
-  const address = event.params.swapAddress;
-  const addressString = address.toHexString();
-  // Don't track unused/broken pools
-  if (BROKEN_POOLS.has(addressString)) {
-    return false;
+// createPoolFromAddress will create a pool in the DB but won't subscribe to it via template
+// because it should already be in the yaml.
+function createPoolFromAddress(address: Address): LiquidityPool {
+  const poolData = POOL_DATA.get(
+    prefixID(dataSource.network(), address.toHexString())
+  );
+
+  return createPool(address, poolData.createdBlockNumber, poolData.createdTimestamp, null);
+}
+
+// createPoolFromEvent will create a pool from a PairCreated event, and subscribe to events from it.
+export function createPoolFromFactoryEvent(event: NewSwapPool): void {
+  const poolAddr = event.params.swapAddress;
+  if (BROKEN_POOLS.has(poolAddr.toHexString())) {
+    return;
+  }
+
+  const pool = LiquidityPool.load(poolAddr.toHexString());
+  if (pool) {
+    return;
+  }
+
+
+  createPool(poolAddr, event.block.number, event.block.timestamp, event.params.pooledTokens)
+  SwapTemplate.create(poolAddr);
+}
+
+// createPoolFromRegistryEvent will create a pool if doesn't exist already when added to the pool registry.
+// This should catch pools deployed manually and not via a deployer.
+export function createPoolFromRegistryEvent(address: Address, block: ethereum.Block): void {
+  if (BROKEN_POOLS.has(address.toHexString())) {
+    return;
+  }
+
+  const pool = LiquidityPool.load(address.toHexString());
+  if (pool) {
+    return;
   }
+
+  createPool(address, block.number, block.timestamp, null);
+  SwapTemplate.create(address);
+}
+
+function createPool(
+  swapAddress: Address, 
+  blockNum: BigInt,
+  timestamp: BigInt, 
+  pooledTokens: Address[] | null
+): LiquidityPool {
+  const address = swapAddress;
+  const addressString = address.toHexString();
+
   const contract = Swap.bind(address);
   let lpTokenAddress = contract.swapStorage().value6;
   // Check if LP token exists
   if (!checkValidToken(lpTokenAddress)) {
     const v1contract = SwapV1.bind(address);
     lpTokenAddress = v1contract.swapStorage().value7;
     if (!checkValidToken(lpTokenAddress)) {
-      log.error("Invalid LP token address {} in pool {}", [
+      log.critical("Invalid LP token address {} in pool {}", [
```

Comment:
we want to terminate the syncing if this happens?

Reply:
Hmm, maybe not. I actually copied this from here: https://github.com/messari/subgraphs/pull/1186/files/53cb5cfddd4ef01fe3e589aa050a739e5c7b7a6c#diff-32d490cbe73170a2dc29bdba3bda4e19b4ca56bfc23bef0d8793b80b9a3ddd2cL500

The functionality was duplicated so I consolidated it into one function. In one place it logged an error and in the other it was a critical 😅 

But yes, might be safer in this case to keep it as an error. Changed it back! (it will now only panic when it fails if creating it from config, as it was initially)